### PR TITLE
Fix crash on some channels

### PR DESCRIPTION
**Pull request by gipawu**
Credits micclass from https://www.vdr-portal.de/forum/index.php?thread/134273-kodi-19-und-pvdr-vnsi-probleme (german forum)

### CODE REVIEW

File: src/InputstreamDemux.cpp [hunk 1]
```diff
@@ -386,8 +386,11 @@ void cVNSIDemux::StreamChange(cResponsePacket* resp)
       const char* language = resp->extract_String();
       uint32_t composition_id(resp->extract_U32());
       uint32_t ancillary_id(resp->extract_U32());
-      props.SetLanguage(language);
-      props.SetSubtitleInfo((composition_id & 0xffff) | ((ancillary_id & 0xffff) << 16));
+      if (strlen(language))
```
Comment (phunkyfish) [inline src/InputstreamDemux.cpp]: Always qualify standard functions: `std::strlen`

**Review by phunkyfish — CHANGES_REQUESTED** [review]
Thanks for contributing!

Once this is ready you will also need to update the `addon.xml.in` and the `changelog.txt` so it can be released.

Also as this fixes a crash a backport PR for Matrix would also be required.

File: src/InputstreamDemux.cpp (hunk 1 above)
Comment (ksooo) [reply src/InputstreamDemux.cpp]: Is the problem actually the string length or rather `language` being `nullptr`? If the latter, the minimal fix would be to change `if (strlen(language))` to `if (language)`.

Comment (gipawu) [reply src/InputstreamDemux.cpp]: The problem is indeed caused by an empty string

File: src/InputstreamDemux.cpp [hunk 2]
```diff
@@ -386,8 +386,11 @@ void cVNSIDemux::StreamChange(cResponsePacket* resp)
       const char* language = resp->extract_String();
       uint32_t composition_id(resp->extract_U32());
       uint32_t ancillary_id(resp->extract_U32());
-      props.SetLanguage(language);
-      props.SetSubtitleInfo((composition_id & 0xffff) | ((ancillary_id & 0xffff) << 16));
+      if (std::strlen(language))
```
Comment (phunkyfish) [inline src/InputstreamDemux.cpp]: Strictly speaking if this is checking numeric length it should be: `std::strlen(language) > 0`

Sorry for not highlighting this on the first review.

**Review by phunkyfish — CHANGES_REQUESTED** [review]

File: src/InputstreamDemux.cpp (hunk 2 above)
Comment (gipawu) [reply src/InputstreamDemux.cpp]: It should be fine now :)

File: src/InputstreamDemux.cpp [hunk 3]
```diff
@@ -386,8 +386,11 @@ void cVNSIDemux::StreamChange(cResponsePacket* resp)
       const char* language = resp->extract_String();
       uint32_t composition_id(resp->extract_U32());
       uint32_t ancillary_id(resp->extract_U32());
-      props.SetLanguage(language);
-      props.SetSubtitleInfo((composition_id & 0xffff) | ((ancillary_id & 0xffff) << 16));
+      if (std::strlen(language) > 0)
```
Comment (ksooo) [inline src/InputstreamDemux.cpp]: Well, looking at the implementation of `PVRStreamProperties::SetLanguage` (https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/pvr/Stream.h#L157) we learn that it expects a 3-letter language id, checks for string not longer than 3 letters, but has no check for string length zero, 1 or 2. So, passing a string with a length less than 3 will lead to a crash. 

@AlwinEsch I think you should fix the implementation of `PVRStreamProperties::SetLanguage`
@gipawu the right workaround for the above mentioned bug in Kodi core is `if (std::strlen(language) == 3)`

**Review by ksooo — CHANGES_REQUESTED** [review]

**Review by ksooo — APPROVED** [review]

**Review by phunkyfish — APPROVED** [review]
Thanks, could you also create the backport on the Matrix branch?

I’ll release this tomorrow.

**Comment by gipawu** [comment]
> Thanks, could you also create the backport on the Matrix branch?
> 
> I’ll release this tomorrow.

Backport PR created.

**Comment by phunkyfish** [comment]
Thanks again for the contributions!!! Releasing both now.

**Pull request merged**